src/components/Pagination/Paginator.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PaginationRow from './PaginationRow';
+import { bindMethods, noop } from '../../common/helpers';
+import { PAGINATION_VIEW_TYPES } from './constants';
+
+class Paginator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    bindMethods(this, ['handleFormSubmit']);
+
+    this.initPagination(props);
+
+    this.state = {
+      pagination: props.pagination,
```

Comment:
Where is the `pagination` state used? IMHO there's no need to keep it in state.

Reply:
just some more background...to be honest, i prefer that state to only be in the wrapper (within mocks) and would prefer the consumption pattern exposed there rather than introduce a stateful component, but I think we can expose the stateful component if that makes consumption easier for some. The convention for `pagination` settings as an object was just following Reactabular: https://reactabular.js.org/#/data/pagination. The convention for exposing all API callbacks for a Table was here: https://material-ui-next.com/demos/tables/ (but I think I can find several more which expose all API callbacks). It is just API preference though...

Good call on removing it from state here though... will update 👍

Reply:
done